lib/pharos/phases/gather_facts.rb

```diff
@@ -98,39 +98,28 @@ def private_interface_address(interface)
       end
 
       # @return [Array<String>]
-      def read_resolvconf_nameservers
-        nameservers = []
-
-        @ssh.file('/etc/resolv.conf').each_line do |line|
-          if match = line.match(/nameserver (.+)/)
-            nameservers << match[1]
-          end
-        end
-
-        nameservers
+      def resolvconf_nameservers
+        @ssh.file('/etc/resolv.conf').lines.map { |l| l[/^nameserver ([\h:.]+)/, 1] }.compact
       end
 
-      LOCALNET = IPAddr.new('127.0.0.0/8')
-
       # Host /etc/resolv.conf is configured to use a nameserver at localhost in the host network namespace
       # @return [Boolean]
-      def check_resolvconf_nameserver_localhost
-        resolvers = read_resolvconf_nameservers.map{ |ip| IPAddr.new(ip) }
-        resolvers.any? { |ip| LOCALNET.include?(ip) }
+      def resolvconf_nameserver_localhost?
+        resolvconf_nameservers.any? { |ip| IPAddr.new(ip).loopback? }
```

Comment:
[`IPAddr#loopback?`](https://ruby-doc.org/stdlib-2.5.1/libdoc/ipaddr/rdoc/IPAddr.html#method-i-loopback-3F) checks for `127.0.0.0/8` + `::1`

Reply:
aaaand it came in ruby 2.5